gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/editor/bottomsheet/GravatarQuickEditorBottomSheet.kt

```diff
@@ -132,53 +137,73 @@ private fun GravatarModalBottomSheet(
         }
     }
 
-    GravatarTheme {
-        ModalBottomSheet(
-            state = modalBottomSheetState,
-        ) {
-            Scrim(
-                modifier = Modifier.clickable { modalBottomSheetState.currentDetent = Hidden },
-                scrimColor = MaterialTheme.colorScheme.scrim.copy(alpha = 0.32f),
-            )
-            Sheet(
-                modifier = Modifier
-                    .clip(RoundedCornerShape(topStart = 28.dp, topEnd = 28.dp))
-                    .background(MaterialTheme.colorScheme.surfaceColorAtElevation(1.dp))
-                    .widthIn(max = 640.dp)
-                    .fillMaxWidth()
-                    .padding(
-                        WindowInsets.navigationBars
-                            .only(WindowInsetsSides.Vertical)
-                            .asPaddingValues(),
-                    ),
+    val configuration = Configuration(LocalConfiguration.current).apply {
+        uiMode = when (colorScheme) {
+            GravatarColorScheme.LIGHT -> Configuration.UI_MODE_NIGHT_NO
+            GravatarColorScheme.DARK -> Configuration.UI_MODE_NIGHT_YES
+            GravatarColorScheme.SYSTEM -> uiMode
+        }
+    }
+    CompositionLocalProvider(
+        LocalConfiguration provides configuration,
+    ) {
+        GravatarTheme {
+            ModalBottomSheet(
+                state = modalBottomSheetState,
             ) {
-                val window = LocalModalWindow.current
-                val isDarkTheme = isSystemInDarkTheme()
-                LaunchedEffect(Unit) {
-                    window.navigationBarColor = Color.TRANSPARENT
-                    WindowInsetsControllerCompat(window, window.decorView).isAppearanceLightNavigationBars =
-                        !isDarkTheme
-                }
-                Surface(
-                    modifier = Modifier
-                        .fillMaxWidth(),
-                    tonalElevation = 1.dp,
+                // Modal content must be taking the uiMode from Activity and doesn't respect the above set CompositionLocalProvider
+                CompositionLocalProvider(
+                    LocalConfiguration provides configuration,
                 ) {
```

Comment:
If I only keep this here, without the one at the top the `QETopBar` and the navigation bars are not properly themed. If I remove this one here, the content of the BottomSheet is not themed. 

That's why we need it in both places. To be honest I'm quite puzzled why the `QETopBar` is not themed because it's part of the content of the BottomSheet...

Reply:
🤯 I've been trying to understand what happens there. No idea. I'll try again tomorrow with the brain fresh.